cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -80,7 +86,7 @@ func indexMachineByProviderID(obj interface{}) ([]string, error) {
 		return nil, nil
 	}
 
-	providerID, found, err := unstructured.NestedString(u.Object, "spec", "providerID")
+	providerID, found, err := unstructured.NestedString(u.UnstructuredContent(), "spec", "providerID")
```

Comment:
Not directly, I think I originally hit an issue while building out the tests where `u.Object` was nil, and it seemed like it would be good to guard against that in general.

Reply:
I'm happy to break this change out separately if you prefer

Reply:
hmm, i'm torn. on one hand it makes it easier to follow the unstructured stuff if we keep it separate, and on the other this is a small change to pick up a few more references.

honestly, my question was just to increase my understanding of the code here. i guess i'm ok to leave it here if no one else has issues.